File: WorldFeed/Server/src/Services/Upload/WorldFeed.Upload.API/Application/Parameters/AddMediaMetadataParameters.cs
namespace WorldFeed.Upload.Application.Parameters
{
    using Newtonsoft.Json;

    using WorldFeed.Common.Public.Parameters;
    using WorldFeed.Upload.API.JsonConverters;
    using WorldFeed.Upload.Models.DTO;

    /// <summary>
    /// For more description visit : https://dev.twitter.com/en/docs/media/upload-media/api-reference/post-media-metadata-create
    /// </summary>
    public interface IAddMediaMetadataParameters : IMediaMetadata, ICustomRequestParameters
    {
    }
    
    public class AddMediaMetadataParameters : CustomRequestParameters, IAddMediaMetadataParameters
    {
        public AddMediaMetadataParameters(long? mediaId)
        {
            this.MediaId = mediaId;
        }
        
        [JsonProperty("media_id")]
        public long? MediaId { get; set; }
        
        [JsonProperty("alt_text")]
        [JsonConverter(typeof(JsonUploadMetadataAltTextConverter))]
        public string AltText { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Parameters/Upload/AddMediaMetadataParameters.ts
import {CustomRequestParameters, ICustomRequestParameters} from "../CustomRequestParameters";
import {IMediaMetadata} from "../../Models/Interfaces/DTO/IMediaMetadata";

// For more description visit : https://dev.twitter.com/en/docs/media/upload-media/api-reference/post-media-metadata-create
export interface IAddMediaMetadataParameters extends IMediaMetadata, ICustomRequestParameters {
}

export class AddMediaMetadataParameters extends CustomRequestParameters implements IAddMediaMetadataParameters {
  constructor(mediaId?: number) {
    super();

    this.mediaId = mediaId;
  }

  // [JsonProperty("media_id")]
  public mediaId?: number;

  // [JsonProperty("alt_text")]
  // [JsonConverter(typeof(JsonUploadMetadataAltTextConverter))]
  public altText: string;
}
